sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2665,6 +2665,23 @@ object SQLConf {
       .checkValue(_ > 0, "The difference must be positive.")
       .createWithDefault(4)
 
+  val NOT_IN_SUBQUERY_SINGLE_COLUMN_OPTIMIZE_ENABLED =
+    buildConf("spark.sql.notInSubquery.singleColumn.optimize.enabled")
+      .internal()
+      .doc("When true, single column not in subquery execution in BroadcastNestedLoopJoinExec " +
+        "will be optimized from M*N calculation into M*log(N) calculation using HashMap lookup " +
```

Comment:
N00b/dumb question: Why M*log(N) instead of M * 1 ? Shouldn't HT probe lookup be O(1) and O(log N).

Reply:
done.